test/e2e/app/app.go

```diff
@@ -427,6 +433,9 @@ func (app *Application) ProcessProposal(_ context.Context, req *abci.RequestProc
 // key/value store ("extensionSum") with the sum of all of the numbers collected
 // from the vote extensions.
 func (app *Application) ExtendVote(_ context.Context, req *abci.RequestExtendVote) (*abci.ResponseExtendVote, error) {
+	app.mu.Lock()
+	defer app.mu.Unlock()
```

Comment:
Here, it'd make more sense since we access `app`'s height, although we only read it -- "int" reads tend to be atomic in most architectures.

Anyway, your change is welcome to not have to reflect on low-level concurrent access to variables (so your change improves readability IMO)

Reply:
> Here, it'd make more sense since we access app's height, although we only read it -- "int" reads tend to be atomic in most architectures.

Concurrent mutating access to a value in Go without explicit synchronization is a memory model violation regardless of the underlying architecture.